From 103a335efbe6dafea82819bfd0abed9e55b83143 Mon Sep 17 00:00:00 2001
From: Chen Gang G <gang.g.chen@intel.com>
Date: Wed, 13 Oct 2021 15:27:26 +0800
Subject: [PATCH 02/54] Kernelflinger: Disalbe UI display function

This can reduce 5 seconds boot delay in un-secure boot case

Tracked-On: OAM-108914
Signed-off-by: Chen Gang G <gang.g.chen@intel.com>
---
 Android.mk      | 1 -
 avb/Android.mk  | 1 -
 kernelflinger.c | 6 +++---
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/Android.mk b/Android.mk
index 5e72383..cfe1bdc 100644
--- a/Android.mk
+++ b/Android.mk
@@ -90,7 +90,6 @@ ifneq ($(TARGET_BUILD_VARIANT),user)
 endif
 
 ifneq ($(strip $(KERNELFLINGER_USE_UI)),false)
-    KERNELFLINGER_CFLAGS += -DUSE_UI
 endif
 
 ifeq ($(KERNELFLINGER_OS_SECURE_BOOT),true)
diff --git a/avb/Android.mk b/avb/Android.mk
index 4a85fa0..22c6008 100644
--- a/avb/Android.mk
+++ b/avb/Android.mk
@@ -68,7 +68,6 @@ LOCAL_STATIC_LIBRARIES := \
 	libkernelflinger-$(TARGET_BUILD_VARIANT)
 
 ifneq ($(strip $(KERNELFLINGER_USE_UI)),false)
-    LOCAL_CFLAGS += -DUSE_UI
 endif
 
 LOCAL_SRC_FILES := \
diff --git a/kernelflinger.c b/kernelflinger.c
index dba41f8..8e26cb2 100644
--- a/kernelflinger.c
+++ b/kernelflinger.c
@@ -1109,7 +1109,7 @@ static VOID enter_fastboot_mode(UINT8 boot_state)
 
 	die();
 }
-static void bootloader_recover_mode(UINT8 boot_state)
+static void bootloader_recover_mode(UINT8 boot_state __unused)
 {
 	enum boot_target target;
 
@@ -1136,8 +1136,8 @@ static void bootloader_recover_mode(UINT8 boot_state)
 	die();
 }
 
-static VOID boot_error(enum ux_error_code error_code, UINT8 boot_state,
-			UINT8 *hash, UINTN hash_size)
+static VOID boot_error(enum ux_error_code error_code __unused, UINT8 boot_state,
+			UINT8 *hash __unused, UINTN hash_size __unused)
 {
 	BOOLEAN power_off = FALSE;
 	enum boot_target bt;
-- 
2.25.1

